Build/libHttpClient.GDK.props

```diff
@@ -96,6 +96,15 @@
     </Link>
   </ItemDefinitionGroup>
 
+  <PropertyGroup>
+    <HCNoZlib>false</HCNoZlib>
+  </PropertyGroup>
+  <ItemDefinitionGroup Condition="'$(HCNoZlib)'=='true'">
+    <ClCompile>
+      <PreprocessorDefinitions>%(PreprocessorDefinitions);HC_NOZLIB</PreprocessorDefinitions>
+    </ClCompile>
+  </ItemDefinitionGroup>
```

Comment:
This also feels like the wrong spot for this.  This props files defines the build config for all LHC projects, not just libHttpClient.lib (and is actually used by PlayFab and XSAPI as well).  This also probably belongs in libHttpClient.Zlib.vcxitems.

Reply:
I think these couldn't live in libHttpClient.Zlib.vcxitems because this property group and item definition group are the ones that define if the libHttpClient.Zlib.vcxitems file will be included in the solution. HC_NOZLIB should be defined somewhere outside libHttpClient.Zlib.vcxitems for the condition below:

https://github.com/microsoft/libHttpClient/blob/36e12bc0089a7347a9a7eb60f0c14d3858ecf9d2/Build/libHttpClient.143.Win32.C/libHttpClient.143.Win32.C.vcxproj#L13C3-L13C136

Reply:
Understood. Since there's actually separate things being defined here, I think we should split this discussion up for clarity.

The 'HCNoZlib' build property shouldn't be defined here.  I don't think you'll ever need to set that.  That should only be set by clients who are building LHC by source. LHC projects should look for that property being set to 'true' and react accordingly.

I'd like to restrict defining preprocessor macro HC_NOZLIB to the LHC lib projects (I believe thats the only place it's needed?) Could that be moved to libHttpClient.Common.vcxitems?

Reply:
Fixed, I removed setting the MSBuild property and moved setting the preprocessor definition to libHttpClient.Common.vcxitems file